src/main/java/eu/europeana/metis/sandbox/domain/Record.java

```diff
@@ -22,21 +21,21 @@
   private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
 
   private final String recordId;
-  private final String datasetId;
+  private final Integer datasetId;
   private final String datasetName;
   private final Country country;
   private final Language language;
   private final byte[] content;
   private String contentString;
 
-  private Record(String recordId, String datasetId, String datasetName,
+  private Record(String recordId, Integer datasetId, String datasetName,
       Country country, Language language, byte[] content) {
     this.recordId = recordId;
     this.datasetId = datasetId;
     this.datasetName = datasetName;
     this.country = country;
     this.language = language;
-    this.content = content;
+    this.content = content; //NOSONAR
```

Comment:
Sonar complains because 

> Mutable members should not be stored or returned directly

But I am creating a copy of the content in the builder, so here we will have a copy of content

Reply:
Same as above: use the suppressed warnings annotation.